docker/client.py

```diff
@@ -239,7 +241,7 @@ def _multiplexed_response_stream_helper(self, response):
         # Disable timeout on the underlying socket to prevent
         # Read timed out(s) for long running processes
         socket = self._get_raw_response_socket(response)
-        if six.PY3:
+        if six.PY3 and hasattr(socket, "_sock"):
```

Comment:
Shouldn't we use the same test in both cases? At first glance it seems checking for existence of the `_sock` attribute would be more robust than basing it upon the value of a publicly accessible high-level attribute like `Client.base_url`.

What was the intent behind making those different?